apps/anon-message-client/src/app/page.tsx

```diff
@@ -124,11 +120,16 @@ export default function () {
 
   useEffect(() => {
     if (!topicDataRaw) return;
-    const topicDataEncoded = Buffer.from(topicDataRaw, "base64");
-    const topicData = JSON.parse(
-      decodeURIComponent(topicDataEncoded.toString("utf-8"))
-    );
-    setTopicData(topicData);
+    const onlyDigits = /^\d+$/;
+    if (onlyDigits.test(topicDataRaw)) {
+      redirect(`/${topicDataRaw}`);
```

Comment:
non-blocking: I don't really like how we check this type based on whether it's a numeric string. An alternative would be adding a type interface for topicData

export enum PayloadType {
  TopicData = "topic-data",
  NullifierHash = "nullifier-hash"
}

export type TopicDataPayload = {
  type: PayloadType.TopicData;
  value: {
    topicId: string;
    chatId: string;
};

export type NullifierHashPayload = {
  type: PayloadType.TopicData;
  value: string
};

export type AnonWebAppPayload = TopicDataPayload | NullifierHashPayload;